install.sh

```diff
@@ -407,6 +420,36 @@ install_airgap_tarball() {
     fi
 }
 
+# install_dev_rpm orchestates the installation of RKE2 unsigned development rpms
+install_dev_rpm() {
+    rpm_installer="${1:-yum}"
+    distro="${2:-centos7}"
+
+    [ -z "${TMP_DIR}" ] && setup_tmp
+
+    case "${rpm_installer}" in
+        *zypper*)
+            ${rpm_installer} install --allow-unsigned-rpm -y \
+                "${STORAGE_URL}/rke2-common-${INSTALL_RKE2_COMMIT}.${distro}.rpm" \
+                "${STORAGE_URL}/rke2-${INSTALL_RKE2_TYPE}-${INSTALL_RKE2_COMMIT}.${distro}.rpm"
+            ;;
+        *)
+            download_dev_rpm "${TMP_DIR}/rke2-common-${INSTALL_RKE2_COMMIT}.${distro}.rpm"
+            download_dev_rpm "${TMP_DIR}/rke2-${INSTALL_RKE2_TYPE}-${INSTALL_RKE2_COMMIT}.${distro}.rpm"
+            ${rpm_installer} install -y  "${TMP_DIR}"/*.rpm
+            ;;
+    esac
+
+    if [ -f "${INSTALL_RKE2_ARTIFACT_PATH}/rke2-images.${SUFFIX}.tar.zst" ]; then
+        return
+    fi
+
+    download_airgap_checksums
```

Comment:
I am sorry  dont really get why do we install tarball if we already installing rpms

Reply:
Yeah, it's uncanny to mix both installation processes I agree. But, worth pointing out we are installing only the airgap images which is not currently handled by the rpms.

The reason behind installing the airgap images from tarballs is that without them rke2 process will fail once started. The error messages are similar to these:

```sh
time="2022-03-23T21:28:04Z" level=info msg="Checking local image archives in /var/lib/rancher/rke2/agent/images for index.docker.io/rancher/rke2-runtime:v1.23.5-dev-d4807018"
time="2022-03-23T21:28:04Z" level=warning msg="Failed to load runtime image index.docker.io/rancher/rke2-runtime:v1.23.5-dev-d4807018 from tarball: no local image available for index.docker.io/rancher/rke2-runtime:v1.23.5-dev-d4807018"
time="2022-03-23T21:28:04Z" level=info msg="Checking local image archives in /var/lib/rancher/rke2/agent/images for index.docker.io/rancher/rke2-runtime:v1.23.5-dev-d4807018"
time="2022-03-23T21:28:04Z" level=warning msg="Failed to load runtime image index.docker.io/rancher/rke2-runtime:v1.23.5-dev-d4807018 from tarball: no local image available for index.docker.io/rancher/rke2-runtime:v1.23.5-dev-d4807018"
time="2022-03-23T21:28:04Z" level=info msg="Pulling runtime image index.docker.io/rancher/rke2-runtime:v1.23.5-dev-d4807018"
time="2022-03-23T21:28:04Z" level=warning msg="Failed to get image from endpoint: GET https://index.docker.io/v2/rancher/rke2-runtime/manifests/v1.23.5-dev-d4807018: MANIFEST_UNKNOWN: manifest unknown"
time="2022-03-23T21:28:04Z" level=fatal msg="failed to get runtime image index.docker.io/rancher/rke2-runtime:v1.23.5-dev-d4807018-dirty: all endpoints failed: GET https://index.docker.io/rancher/rke2-runtime:v1.23.5-dev-d4807018"
```

By looking at [bootstrap.go](https://github.com/rancher/rke2/blob/master/pkg/bootstrap/bootstrap.go#L130), I found that rke2 will look for the `rke2-runtime` image first in the host file system then try to pull the image from the configured image registry. Since we don't publish rke2-runtime images per each commit, I thought it was simpler to install the rke2-airgap images tarball right after the rpm.